crash on clear_board twice

When I try to open a sgf file via GoGUI, michi-c2 crashes. Then I traced the issue is caused by two consequtive `clear_board`. To reproduce:
````
$ ./michi gtp
clear_board
Move: 0     Black: 0 caps   White: 0 caps   Komi: 7.5
 19 . . . . . . . . . . . . . . . . . . .     . . . . . . . . . . . . . . . . . . .
 18 . . . . . . . . . . . . . . . . . . .     . . . . . . . . . . . . . . . . . . .
 17 . . . . . . . . . . . . . . . . . . .     . . . . . . . . . . . . . . . . . . .
 16 . . . . . . . . . . . . . . . . . . .     . . . . . . . . . . . . . . . . . . .
 15 . . . . . . . . . . . . . . . . . . .     . . . . . . . . . . . . . . . . . . .
 14 . . . . . . . . . . . . . . . . . . .     . . . . . . . . . . . . . . . . . . .
 13 . . . . . . . . . . . . . . . . . . .     . . . . . . . . . . . . . . . . . . .
 12 . . . . . . . . . . . . . . . . . . .     . . . . . . . . . . . . . . . . . . .
 11 . . . . . . . . . . . . . . . . . . .     . . . . . . . . . . . . . . . . . . .
 10 . . . . . . . . . . . . . . . . . . .     . . . . . . . . . . . . . . . . . . .
 9  . . . . . . . . . . . . . . . . . . .     . . . . . . . . . . . . . . . . . . .
 8  . . . . . . . . . . . . . . . . . . .     . . . . . . . . . . . . . . . . . . .
 7  . . . . . . . . . . . . . . . . . . .     . . . . . . . . . . . . . . . . . . .
 6  . . . . . . . . . . . . . . . . . . .     . . . . . . . . . . . . . . . . . . .
 5  . . . . . . . . . . . . . . . . . . .     . . . . . . . . . . . . . . . . . . .
 4  . . . . . . . . . . . . . . . . . . .     . . . . . . . . . . . . . . . . . . .
 3  . . . . . . . . . . . . . . . . . . .     . . . . . . . . . . . . . . . . . . .
 2  . . . . . . . . . . . . . . . . . . .     . . . . . . . . . . . . . . . . . . .
 1  . . . . . . . . . . . . . . . . . . .     . . . . . . . . . . . . . . . . . . .
    A B C D E F G H J K L M N O P Q R S T 

= 

clear_board
michi(27531,0x7fff78670300) malloc: *** error for object 0x7f93f8405070: pointer being freed was not allocated
*** set a breakpoint in malloc_error_break to debug
Abort trap: 6
````

I have also compiled and tested the code on Ubuntu 14.04 without any error:

```
$ make
gcc  -DNDEBUG -O3 -march=native -msse4 -fshort-enums -Wall -Wno-char-subscripts -c -std=gnu99 ui.c 
gcc  -DNDEBUG -O3 -march=native -msse4 -fshort-enums -Wall -Wno-char-subscripts -c -std=gnu99 sgf.c 
gcc  -DNDEBUG -O3 -march=native -msse4 -fshort-enums -Wall -Wno-char-subscripts -c -std=gnu99 control.c 
gcc  -DNDEBUG -O3 -march=native -msse4 -fshort-enums -Wall -Wno-char-subscripts -c -std=gnu99 michi.c 
gcc  -DNDEBUG -O3 -march=native -msse4 -fshort-enums -Wall -Wno-char-subscripts -c -std=gnu99 params.c 
gcc  -DNDEBUG -O3 -march=native -msse4 -fshort-enums -Wall -Wno-char-subscripts -c -std=gnu99 board.c 
gcc  -DNDEBUG -O3 -march=native -msse4 -fshort-enums -Wall -Wno-char-subscripts -c -std=gnu99 board_util.c 
gcc  -DNDEBUG -O3 -march=native -msse4 -fshort-enums -Wall -Wno-char-subscripts -c -std=gnu99 patterns.c 
patterns.c: In function ‘init_large_patterns’:
patterns.c:651:50: warning: ‘fspat’ may be used uninitialized in this function [-Wmaybe-uninitialized]
     if (verbosity > 0 && (fprob == NULL || fspat == NULL)) {
                                                  ^
gcc  -DNDEBUG -O3 -march=native -msse4 -fshort-enums -Wall -Wno-char-subscripts -c -std=gnu99 debug.c 
gcc  -DNDEBUG -O3 -march=native -msse4 -fshort-enums -Wall -Wno-char-subscripts -c -std=gnu99 main.c 
gcc -DNDEBUG -O3 -march=native -msse4 -fshort-enums -Wall -Wno-char-subscripts -std=gnu99 -o michi ui.o sgf.o control.o michi.o params.o board.o board_util.o patterns.o debug.o main.o -lm
$ ./michi gtp
Warning: michi cannot load pattern files, It will be much weaker. Consider lowering EXPAND_VISITS 8->2
clear_board
Move: 0     Black: 0 caps   White: 0 caps   Komi: 7.5
 19 . . . . . . . . . . . . . . . . . . .     . . . . . . . . . . . . . . . . . . .
 18 . . . . . . . . . . . . . . . . . . .     . . . . . . . . . . . . . . . . . . .
 17 . . . . . . . . . . . . . . . . . . .     . . . . . . . . . . . . . . . . . . .
 16 . . . . . . . . . . . . . . . . . . .     . . . . . . . . . . . . . . . . . . .
 15 . . . . . . . . . . . . . . . . . . .     . . . . . . . . . . . . . . . . . . .
 14 . . . . . . . . . . . . . . . . . . .     . . . . . . . . . . . . . . . . . . .
 13 . . . . . . . . . . . . . . . . . . .     . . . . . . . . . . . . . . . . . . .
 12 . . . . . . . . . . . . . . . . . . .     . . . . . . . . . . . . . . . . . . .
 11 . . . . . . . . . . . . . . . . . . .     . . . . . . . . . . . . . . . . . . .
 10 . . . . . . . . . . . . . . . . . . .     . . . . . . . . . . . . . . . . . . .
 9  . . . . . . . . . . . . . . . . . . .     . . . . . . . . . . . . . . . . . . .
 8  . . . . . . . . . . . . . . . . . . .     . . . . . . . . . . . . . . . . . . .
 7  . . . . . . . . . . . . . . . . . . .     . . . . . . . . . . . . . . . . . . .
 6  . . . . . . . . . . . . . . . . . . .     . . . . . . . . . . . . . . . . . . .
 5  . . . . . . . . . . . . . . . . . . .     . . . . . . . . . . . . . . . . . . .
 4  . . . . . . . . . . . . . . . . . . .     . . . . . . . . . . . . . . . . . . .
 3  . . . . . . . . . . . . . . . . . . .     . . . . . . . . . . . . . . . . . . .
 2  . . . . . . . . . . . . . . . . . . .     . . . . . . . . . . . . . . . . . . .
 1  . . . . . . . . . . . . . . . . . . .     . . . . . . . . . . . . . . . . . . .
    A B C D E F G H J K L M N O P Q R S T 

= 

clear_board
Move: 0     Black: 0 caps   White: 0 caps   Komi: 7.5
 19 . . . . . . . . . . . . . . . . . . .     . . . . . . . . . . . . . . . . . . .
 18 . . . . . . . . . . . . . . . . . . .     . . . . . . . . . . . . . . . . . . .
 17 . . . . . . . . . . . . . . . . . . .     . . . . . . . . . . . . . . . . . . .
 16 . . . . . . . . . . . . . . . . . . .     . . . . . . . . . . . . . . . . . . .
 15 . . . . . . . . . . . . . . . . . . .     . . . . . . . . . . . . . . . . . . .
 14 . . . . . . . . . . . . . . . . . . .     . . . . . . . . . . . . . . . . . . .
 13 . . . . . . . . . . . . . . . . . . .     . . . . . . . . . . . . . . . . . . .
 12 . . . . . . . . . . . . . . . . . . .     . . . . . . . . . . . . . . . . . . .
 11 . . . . . . . . . . . . . . . . . . .     . . . . . . . . . . . . . . . . . . .
 10 . . . . . . . . . . . . . . . . . . .     . . . . . . . . . . . . . . . . . . .
 9  . . . . . . . . . . . . . . . . . . .     . . . . . . . . . . . . . . . . . . .
 8  . . . . . . . . . . . . . . . . . . .     . . . . . . . . . . . . . . . . . . .
 7  . . . . . . . . . . . . . . . . . . .     . . . . . . . . . . . . . . . . . . .
 6  . . . . . . . . . . . . . . . . . . .     . . . . . . . . . . . . . . . . . . .
 5  . . . . . . . . . . . . . . . . . . .     . . . . . . . . . . . . . . . . . . .
 4  . . . . . . . . . . . . . . . . . . .     . . . . . . . . . . . . . . . . . . .
 3  . . . . . . . . . . . . . . . . . . .     . . . . . . . . . . . . . . . . . . .
 2  . . . . . . . . . . . . . . . . . . .     . . . . . . . . . . . . . . . . . . .
 1  . . . . . . . . . . . . . . . . . . .     . . . . . . . . . . . . . . . . . . .
    A B C D E F G H J K L M N O P Q R S T 

= 

clear_board
Move: 0     Black: 0 caps   White: 0 caps   Komi: 7.5
 19 . . . . . . . . . . . . . . . . . . .     . . . . . . . . . . . . . . . . . . .
 18 . . . . . . . . . . . . . . . . . . .     . . . . . . . . . . . . . . . . . . .
 17 . . . . . . . . . . . . . . . . . . .     . . . . . . . . . . . . . . . . . . .
 16 . . . . . . . . . . . . . . . . . . .     . . . . . . . . . . . . . . . . . . .
 15 . . . . . . . . . . . . . . . . . . .     . . . . . . . . . . . . . . . . . . .
 14 . . . . . . . . . . . . . . . . . . .     . . . . . . . . . . . . . . . . . . .
 13 . . . . . . . . . . . . . . . . . . .     . . . . . . . . . . . . . . . . . . .
 12 . . . . . . . . . . . . . . . . . . .     . . . . . . . . . . . . . . . . . . .
 11 . . . . . . . . . . . . . . . . . . .     . . . . . . . . . . . . . . . . . . .
 10 . . . . . . . . . . . . . . . . . . .     . . . . . . . . . . . . . . . . . . .
 9  . . . . . . . . . . . . . . . . . . .     . . . . . . . . . . . . . . . . . . .
 8  . . . . . . . . . . . . . . . . . . .     . . . . . . . . . . . . . . . . . . .
 7  . . . . . . . . . . . . . . . . . . .     . . . . . . . . . . . . . . . . . . .
 6  . . . . . . . . . . . . . . . . . . .     . . . . . . . . . . . . . . . . . . .
 5  . . . . . . . . . . . . . . . . . . .     . . . . . . . . . . . . . . . . . . .
 4  . . . . . . . . . . . . . . . . . . .     . . . . . . . . . . . . . . . . . . .
 3  . . . . . . . . . . . . . . . . . . .     . . . . . . . . . . . . . . . . . . .
 2  . . . . . . . . . . . . . . . . . . .     . . . . . . . . . . . . . . . . . . .
 1  . . . . . . . . . . . . . . . . . . .     . . . . . . . . . . . . . . . . . . .
    A B C D E F G H J K L M N O P Q R S T 

= 

quit
= 
```

Here is the version of code I am using, [on dropbox](https://www.dropbox.com/sh/iexwu5amz85789b/AAC7txldB0wtZYjCHQcjLFbPa?dl=0).

Here is the compile and run log on Mac OS X 10.10.5, with `malloc` error:

```
Air:michi-c2-796eedd horace$ make
gcc  -DNDEBUG -O3 -march=native -msse4 -fshort-enums -Wall -Wno-char-subscripts -c -std=gnu99 ui.c 
gcc  -DNDEBUG -O3 -march=native -msse4 -fshort-enums -Wall -Wno-char-subscripts -c -std=gnu99 sgf.c 
gcc  -DNDEBUG -O3 -march=native -msse4 -fshort-enums -Wall -Wno-char-subscripts -c -std=gnu99 control.c 
gcc  -DNDEBUG -O3 -march=native -msse4 -fshort-enums -Wall -Wno-char-subscripts -c -std=gnu99 michi.c 
gcc  -DNDEBUG -O3 -march=native -msse4 -fshort-enums -Wall -Wno-char-subscripts -c -std=gnu99 params.c 
gcc  -DNDEBUG -O3 -march=native -msse4 -fshort-enums -Wall -Wno-char-subscripts -c -std=gnu99 board.c 
gcc  -DNDEBUG -O3 -march=native -msse4 -fshort-enums -Wall -Wno-char-subscripts -c -std=gnu99 board_util.c 
gcc  -DNDEBUG -O3 -march=native -msse4 -fshort-enums -Wall -Wno-char-subscripts -c -std=gnu99 patterns.c 
gcc  -DNDEBUG -O3 -march=native -msse4 -fshort-enums -Wall -Wno-char-subscripts -c -std=gnu99 debug.c 
gcc  -DNDEBUG -O3 -march=native -msse4 -fshort-enums -Wall -Wno-char-subscripts -c -std=gnu99 main.c 
gcc -DNDEBUG -O3 -march=native -msse4 -fshort-enums -Wall -Wno-char-subscripts -std=gnu99 -o michi ui.o sgf.o control.o michi.o params.o board.o board_util.o patterns.o debug.o main.o -lm
Air:michi-c2-796eedd horace$ ./michi gtp
Warning: michi cannot load pattern files, It will be much weaker. Consider lowering EXPAND_VISITS 8->2
clear_board
Move: 0     Black: 0 caps   White: 0 caps   Komi: 7.5
 19 . . . . . . . . . . . . . . . . . . .     . . . . . . . . . . . . . . . . . . .
 18 . . . . . . . . . . . . . . . . . . .     . . . . . . . . . . . . . . . . . . .
 17 . . . . . . . . . . . . . . . . . . .     . . . . . . . . . . . . . . . . . . .
 16 . . . . . . . . . . . . . . . . . . .     . . . . . . . . . . . . . . . . . . .
 15 . . . . . . . . . . . . . . . . . . .     . . . . . . . . . . . . . . . . . . .
 14 . . . . . . . . . . . . . . . . . . .     . . . . . . . . . . . . . . . . . . .
 13 . . . . . . . . . . . . . . . . . . .     . . . . . . . . . . . . . . . . . . .
 12 . . . . . . . . . . . . . . . . . . .     . . . . . . . . . . . . . . . . . . .
 11 . . . . . . . . . . . . . . . . . . .     . . . . . . . . . . . . . . . . . . .
 10 . . . . . . . . . . . . . . . . . . .     . . . . . . . . . . . . . . . . . . .
 9  . . . . . . . . . . . . . . . . . . .     . . . . . . . . . . . . . . . . . . .
 8  . . . . . . . . . . . . . . . . . . .     . . . . . . . . . . . . . . . . . . .
 7  . . . . . . . . . . . . . . . . . . .     . . . . . . . . . . . . . . . . . . .
 6  . . . . . . . . . . . . . . . . . . .     . . . . . . . . . . . . . . . . . . .
 5  . . . . . . . . . . . . . . . . . . .     . . . . . . . . . . . . . . . . . . .
 4  . . . . . . . . . . . . . . . . . . .     . . . . . . . . . . . . . . . . . . .
 3  . . . . . . . . . . . . . . . . . . .     . . . . . . . . . . . . . . . . . . .
 2  . . . . . . . . . . . . . . . . . . .     . . . . . . . . . . . . . . . . . . .
 1  . . . . . . . . . . . . . . . . . . .     . . . . . . . . . . . . . . . . . . .
    A B C D E F G H J K L M N O P Q R S T 

= 

clear_board
michi(30447,0x7fff78670300) malloc: *** error for object 0x7fd87a404cd0: pointer being freed was not allocated
*** set a breakpoint in malloc_error_break to debug
Abort trap: 6

```

Here is the `michi.log`:

```
I     0/000 michi-c Version 1.4
I     0/000 size of struct Position = 14540 bytes
I     0/000 size of struct TreeNode = 40 bytes
I     0/000 size of int  = 4 bytes
I     0/000 size of long = 8 bytes
T     0/000 time          :  nsims seconds  sims/s moves tleft
S     0/000 search: idum dkomi 1st (2nd) best2  bestr bestwr games/s
I     0/000 Loading pattern probs ...
w     0/000 Cannot load pattern file:patterns.prob
I     0/000 =========== Hashtable initialization synthesis ==========
C     0/000 clear_board
I     1/001 BEGIN GAME 1, random seed = 3698499369
C     1/001 clear_board
```

--- a/ui.c
+++ b/ui.c
@@ -240,14 +240,14 @@ void begin_game(void) {
     log_fmt_s('I', buf, NULL);
 }
 
-char* gtp_clear_board(Game *game, TreeNode *tree)
+char* gtp_clear_board(Game *game, TreeNode **tree)
 {
     char *ret;
 
     if (game_ongoing) begin_game();
     game_ongoing = 0;
-    free_tree(tree);
-    tree = new_tree_node();
+    free_tree(*tree);
+    *tree = new_tree_node();
     nplayouts_per_second = -1.0;
     ret = game_clear_board(game);
     return ret;
@@ -700,7 +700,7 @@ void gtp_io(Game *game, FILE *f, FILE *out, int owner_map[], int score_count[])
         else if (strcmp(command, "boardsize") == 0)
             ret = gtp_boardsize(pos);
         else if (strcmp(command, "clear_board") == 0)
-            ret = gtp_clear_board(game, tree);
+            ret = gtp_clear_board(game, &tree);
         else if (strcmp(command,"debug") == 0)
             ret = debug(game);
         else if (strcmp(command,"final_score") == 0)

